src/main/java/org/folio/orders/events/handlers/ReceiptStatusConsistency.java

```diff
@@ -1,162 +1,85 @@
-            boolean statusUpdated = purchaseOrderLineService.updatePoLineReceiptStatusWithoutSave(poLine, receivingStatus);
-            if (statusUpdated) {
-              purchaseOrderLineService.saveOrderLine(poLine, requestContext)
-                .map(aVoid -> {
-                  // send event to update order status
-                  updateOrderStatus(poLine, okapiHeaders, requestContext);
-                  promise.complete();
-                  return null;
-                })
-                .onFailure(e -> {
-                  logger.error("The error updating poLine by id {}", poLineIdUpdate, e);
-                  promise.fail(e);
-                });
-            } else {
-              promise.complete();
-            }
-            return null;
-          })
-          .onFailure(e -> {
-            logger.error("The error getting poLine by id {}", poLineIdUpdate, e);
-            promise.fail(e);
-          }))
-      .onFailure(e -> {
-        logger.error("The error happened getting all pieces by poLine {}", poLineIdUpdate, e);
-        promise.fail(e);
-      });
 
-    // Now wait for all operations to be completed and send reply
-    completeAllFutures(futures, message);
-  }
+    var poLineId = getPoLineId(messageFromEventBus);
+    var future = pieceStorageService.getPiecesByLineId(poLineId, requestContext)
+      .compose(pieces -> purchaseOrderLineService.getOrderLineById(poLineId, requestContext)
+        .compose(poLine -> updatePoLineAndOrderStatuses(pieces, poLine, requestContext))
+        .onFailure(e -> logger.error("Exception occurred while fetching PoLine by id: '{}'", poLineId, e)))
+      .onFailure(e -> logger.error("Exception occurred while fetching pieces by PoLine id: '{}'", poLineId, e));
 
-  private void updateOrderStatus(PoLine poLine, Map<String, String> okapiHeaders, RequestContext requestContext) {
-    List<JsonObject> poIds = StreamEx
-      .of(poLine)
-      .map(PoLine::getPurchaseOrderId)
-      .distinct()
-      .map(orderId -> new JsonObject().put(ORDER_ID, orderId))
-      .toList();
-    JsonObject messageContent = new JsonObject();
-    messageContent.put(OKAPI_HEADERS, okapiHeaders);
-    // Collect order ids which should be processed
-    messageContent.put(EVENT_PAYLOAD, new JsonArray(poIds));
-    HelperUtils.sendEvent(MessageAddress.RECEIVE_ORDER_STATUS_UPDATE, messageContent, requestContext);
+    completeAllFutures(List.of(future), message);
   }
 
-  private ReceiptStatus calculatePoLineReceiptStatus(PoLine poLine, List<Piece> pieces) {
-
-    if (CollectionUtils.isEmpty(pieces)) {
-      logger.info("No pieces processed - receipt status unchanged for PO Line '{}'", poLine.getId());
-      return poLine.getReceiptStatus();
+  private Future<Void> updatePoLineAndOrderStatuses(List<Piece> pieces, PoLine poLine, RequestContext requestContext) {
+    if (PoLineCommonUtil.isCancelledOrOngoingStatus(poLine)) {
```

Comment:
can improve by adding info log here also

Reply:
Thanks, added bunch of logs